plugins/vpp/srplugin/descriptor/localsid.go

```diff
@@ -0,0 +1,282 @@
+	}
+	if localSID.GetFibTableId() < 0 {
+		return scheduler.NewInvalidValueError(errors.Errorf("fibtableid can't be lower than zero, input value %v", localSID.GetFibTableId()), "fibtableid")
+	}
+
+	// checking end functions
+	switch ef := localSID.EndFunction.(type) {
+	case *srv6.LocalSID_BaseEndFunction:
+	case *srv6.LocalSID_EndFunction_X:
+		_, err := ParseIPv6(ef.EndFunction_X.NextHop)
+		if err != nil {
+			return scheduler.NewInvalidValueError(errors.Errorf("failed to parse next hop %s, should be a valid ipv6 address: %v", ef.EndFunction_X.NextHop, err), "endfunction_X.NextHop")
+		}
+	case *srv6.LocalSID_EndFunction_T:
+	case *srv6.LocalSID_EndFunction_DX2:
+	case *srv6.LocalSID_EndFunction_DX4:
+		_, err := ParseIPv4(ef.EndFunction_DX4.NextHop)
+		if err != nil {
+			return scheduler.NewInvalidValueError(errors.Errorf("failed to parse next hop %s, should be a valid ipv4 address: %v", ef.EndFunction_DX4.NextHop, err), "endfunction_DX4.NextHop")
+		}
+	case *srv6.LocalSID_EndFunction_DX6:
+		_, err := ParseIPv6(ef.EndFunction_DX6.NextHop)
+		if err != nil {
+			return scheduler.NewInvalidValueError(errors.Errorf("failed to parse next hop %s, should be a valid ipv6 address: %v", ef.EndFunction_DX6.NextHop, err), "endfunction_DX6.NextHop")
+		}
+	case *srv6.LocalSID_EndFunction_DT4:
+	case *srv6.LocalSID_EndFunction_DT6:
+	case *srv6.LocalSID_EndFunction_AD:
+		ip := net.ParseIP(ef.EndFunction_AD.ServiceAddress)
+		if ip == nil {
+			return scheduler.NewInvalidValueError(errors.Errorf("failed to parse service address %s, should be a valid ip address(ipv4 or ipv6): %v", ef.EndFunction_AD.ServiceAddress, err), "endfunction_AD.ServiceAddress")
+		}
+	case nil:
+		return scheduler.NewInvalidValueError(errors.New("end function must be provided"), "endfunction")
+	default:
+		return scheduler.NewInvalidValueError(errors.Errorf("end function has unexpected model link type %T", ef), "endfunction")
+	}
+
+	return nil
+}
+
+// Create creates new Local SID into VPP using VPP's binary api
+func (d *LocalSIDDescriptor) Create(key string, value *srv6.LocalSID) (metadata interface{}, err error) {
+	sid, _ := ParseIPv6(value.GetSid()) // already validated
+	if err := d.srHandler.AddLocalSid(sid, value); err != nil {
+		return nil, errors.Errorf("failed to add local sid %s: %v", sid.String(), err)
+	}
+	return nil, nil
+}
+
+// Delete removes Local SID from VPP using VPP's binary api
+func (d *LocalSIDDescriptor) Delete(key string, value *srv6.LocalSID, metadata interface{}) error {
+	sid, _ := ParseIPv6(value.GetSid()) // already validated
+	if err := d.srHandler.DeleteLocalSid(sid); err != nil {
+		return errors.Errorf("failed to delete local sid %s: %v", sid.String(), err)
+	}
+	return nil
+}
+
+// UpdateWithRecreate always returns true - LocalSIDs are always updated via re-creation.
+func (d *LocalSIDDescriptor) UpdateWithRecreate(key string, oldValue, newValue *srv6.LocalSID, metadata interface{}) bool {
+	return true
```

Comment:
With the latest vpp-agent, if the Update operation is not implemented, it is automatically determined to apply any change via re-creation - i.e. it is not needed to define UpdateWithRecreate always returning true.

Reply:
fixed in https://github.com/ligato/vpp-agent/pull/1174/commits/18b2bc5124634cb1720353d988f47e0caea00f18